tests/core/test_indexset.py

```diff
@@ -1,47 +1,105 @@
+        assert indexset_1.id != indexset_2.id
 
     def test_add_elements(self, test_mp):
         run = test_mp.Run("Model", "Scenario", "new")
         test_elements = ["foo", "bar"]
-        index_set_1 = run.optimization.IndexSet("IndexSet 1")
-        index_set_1.add(test_elements)
+        indexset_1 = run.optimization.IndexSet("IndexSet 1")
+        indexset_1.add(test_elements)
         run.optimization.IndexSet("IndexSet 2").add(test_elements)
-        index_set_2 = run.optimization.IndexSet("IndexSet 2")
-        assert index_set_1.elements == index_set_2.elements
+        indexset_2 = run.optimization.IndexSet("IndexSet 2")
+        assert indexset_1.elements == indexset_2.elements
 
         with pytest.raises(ValueError):
-            index_set_1.add(["baz", "foo"])
+            indexset_1.add(["baz", "foo"])
 
         with pytest.raises(ValueError):
-            index_set_2.add(["baz", "baz"])
+            indexset_2.add(["baz", "baz"])
 
-        index_set_1.add(1)
-        index_set_3 = run.optimization.IndexSet("IndexSet 1")
-        index_set_2.add("1")
-        index_set_4 = run.optimization.IndexSet("IndexSet 2")
-        assert index_set_3.elements != index_set_4.elements
-        assert len(index_set_3.elements) == len(index_set_4.elements)
+        indexset_1.add(1)
+        indexset_3 = run.optimization.IndexSet("IndexSet 1")
+        indexset_2.add("1")
+        indexset_4 = run.optimization.IndexSet("IndexSet 2")
+        assert indexset_3.elements != indexset_4.elements
+        assert len(indexset_3.elements) == len(indexset_4.elements)
+
+    def test_list_indexsets(self, test_mp):
+        run = test_mp.Run("Model", "Scenario", "new")
+        # Per default, list() lists only `default` version runs:
+        run.set_as_default()
+        indexset_1 = run.optimization.IndexSet("Indexset 1")
+        indexset_2 = run.optimization.IndexSet("Indexset 2")
+        expected_ids = [indexset_1.id, indexset_2.id]
+        list_ids = [indexset.id for indexset in run.optimization.indexsets.list()]
+        assert not (set(expected_ids) ^ set(list_ids))
+
+        # Test retrieving just one result by providing a name
+        expected_id = [indexset_1.id]
+        list_id = [
+            indexset.id
+            for indexset in run.optimization.indexsets.list(name="Indexset 1")
+        ]
+        assert not (set(expected_id) ^ set(list_id))
+
+    def test_tabulate_indexsets(self, test_mp):
+        run = test_mp.Run("Model", "Scenario", "new")
+        # Per default, tabulate() lists only `default` version runs:
+        run.set_as_default()
+        indexset_1 = run.optimization.IndexSet("Indexset 1")
+        indexset_2 = run.optimization.IndexSet("Indexset 2")
+        exp = df_from_list(indexsets=[indexset_1, indexset_2])
+        res = run.optimization.indexsets.tabulate()
+        # utils.assert_unordered_equality doesn't like lists, so make sure the order in
```

Comment:
What does "doesnt like lists" mean? assert_unordered_equality is meant only for DFs afaik...

Reply:
It is, that's why it didn't work when I tried to pass in a list of DFs. That's why I added this auxiliary `df_from_list()` function. We could rephrase this hint as "... only accepts pd.DataFrames,". Or we remove this comment entirely, but both the order of `indexset_1, indexset_2` as arguments for `df_from_list()` as well as the order of `indexset.name, indexset.elements, ...` in the function's return-value-DF are important for the test to work.